### ISSUE TITLE
strings: No such file or directory

### ISSUE THREAD

**Issue by KurzonDax** [dokterdok/Continuity-Activation-Tool#165]
This is more of an FYI than anything.  When attempting to run the shell script manually from terminal on an early 2011 MBP 17", the following error will occur. **NOTE: the line number is wrong because I inserted some echo's in the script to try and troubleshoot what was going on.**  The same error from strings occurs when checking the WiFi whitelist also, and the script returns you to the main menu.
```
Verifying Bluetooth blacklist status... ./contitool.sh: line 611: ./strings: No such file or directory 
```
I didn't spend a lot of time troubleshooting this, but I suspect it may have something to do with the variable expansions.  I'm not super adept at bash scripting either, so I could be way off.

The good news is the script works flawlessly if you just double-click on the app from finder.

May be worth noting the errors occurred regardless of whether I invoked the script with sudo, or if I let the script invoke sudo on its own.

Really nice work with this though.  Call me impressed :)

**Comment by dokterdok** [comment]
Hi, thanks for the feedback.
It looks like the script can't find the strings utility.
Make sure the strings utility is in the same folder as contitool.sh. A copy is bundled with the app in ```Continuity Activation Tool.app/Contents/Resources```

An alternative is to install the Xcode command-line tools by typing ```xcode-select --install``` in the Terminal, and then modify the following line in contitool.sh:
```stringsPath="$appDir/strings"``` --> ```stringsPath="strings"```

Then you won't need to have the strings utility bundled with the script.

**Comment by ShaveIce** [comment]
Hi, I've had this same problem.  It occurred at lines 605 and 545 where the blacklist and whitelist variables are constructed by calling the strings utility using `${stringsPath}`.  The problem seems to stem from the `appDir` definition at line 23 where `appDir` is derived from the command line argument `$0`.

When I ran ./contitool.sh, appDir was set to "." as expected.  However somewhere in contitool, the current directory was changed before the blacklist and whitelist lines were evaluated (I inserted a `pwd` at line 605 and it showed the current directory as "/Users").

I was able to get the script working by changing line 23 to:
```
appDir=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
```

This was suggested code to get the absolute pathname in the command line under OS X, short of using another language/utility (e.g. python).

Other than that, the tool worked perfectly with both my Early 2011 MBP and Mid 2011 Mac mini.  Thanks for taking the time and effort to create this useful tool.
